File: AudioPlayer2/app/src/main/java/com/music/audioplayer/MyFragmentPagerAdapter.java
package com.music.audioplayer;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.io.File;
import java.util.ArrayList;

public class MyFragmentPagerAdapter extends FragmentPagerAdapter
{
    private FoldersongFragment foldersongFragment;
    private AllsongFragment allsongFragment;
    private PlayListFragment playListFragment;
    private FavouritesongFragment favouritesongFragment;
    private HomeActivity homeActivity;
    private ArrayList<CustomFile> playListFiles;

    public MyFragmentPagerAdapter(FragmentManager fm, HomeActivity homeActivity, ArrayList<CustomFile> playListFiles)
    {
        super(fm);
        this.homeActivity=homeActivity;
        this.playListFiles=playListFiles;
    }



    @Override
    public Fragment getItem(int i)
    {
        Fragment fragment=null;
        switch (i)
        {
            case 0:

                fragment=new FoldersongFragment(homeActivity);
                foldersongFragment= (FoldersongFragment) fragment;
                break;

            case 1:
                fragment=new AllsongFragment(homeActivity);
                allsongFragment=(AllsongFragment) fragment;
                break;
            case 2:
                fragment=new PlayListFragment(homeActivity,playListFiles);
                playListFragment= (PlayListFragment) fragment;
                break;
            case 3:
                fragment=new FavouritesongFragment(homeActivity);
                favouritesongFragment= (FavouritesongFragment) fragment;
                break;
        }
        return fragment;
    }

    @Override
    public int getCount()
    {
        return 4;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position)
    {
        String title=null;

        switch (position)
        {
            case 0:
                title="FOLDER";
                break;
            case 1:
                title="ALL SONGS";
                break;
            case 2:
                title="PLAY LIST";
                break;
            case 3:
                title="FAVOURITE";
                break;

        }
        return title;
    }



    public AllsongFragment getAllsongFragmentObject() {
        return allsongFragment;
    }
    public PlayListFragment getPlayListFragmentObject() {
        return playListFragment;
    }
    public FoldersongFragment getFolderFragmentObject() {
        return foldersongFragment;
    }
    public FavouritesongFragment getFavouriteSongFragmentObject() {
        return favouritesongFragment;
    }


}

File: KotlinDemoApp1/app/src/main/java/com/nisith/kotlindemoapp1/Adapters/MyFragmentPagerAdapter.kt
package com.nisith.kotlindemoapp1.Adapters


import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.nisith.kotlindemoapp1.Fragments.DrumPadFragment
import com.nisith.kotlindemoapp1.Fragments.MusicFragment

@Suppress("DEPRECATION")
class MyFragmentPagerAdapter( fragmentManager: FragmentManager) : FragmentPagerAdapter(fragmentManager) {

    override fun getItem(position: Int): Fragment {
        if (position == 1){
            val drumPadFragment = DrumPadFragment()
            Log.d("MNBVC","fragment "+drumPadFragment)
            return drumPadFragment
        }else{
            return MusicFragment()
        }
    }

    override fun getCount(): Int {
        return 2
    }

    override fun getPageTitle(position: Int): CharSequence? {
        var title: CharSequence = ""
        when(position){
            0 -> {
                title = "DrumPad"
            }
            1 -> {
                title = "All Files"
            }
        }
        return title
    }




}